compile-angel.el

```diff
@@ -411,17 +449,13 @@ FEATURE-NAME is a string representing the feature name being loaded."
        ;; compile-angel. This is important because `.el` files in these
        ;; directories should never be compiled, or Doom may fail to load some of
        ;; them correctly.
-       ((or (and (boundp 'doom-user-dir)
-                 (file-in-directory-p el-file doom-user-dir))
-            (and (boundp 'doom-emacs-dir)
-                 (file-in-directory-p
-                  el-file (expand-file-name "lisp" doom-emacs-dir)))
-            (and (boundp 'doom-modules-dir)
-                 (file-in-directory-p
-                  el-file (expand-file-name doom-modules-dir))))
+       ((and (boundp 'doom-user-dir)
+             (or (string-prefix-p doom-user-dir el-file)
```

Comment:
Using  `string-prefix-p` may be faster, but `file-in-directory-p` is more accurate because it uses `file-truename` to normalize both paths (e.g., paths can differ if they are symlinks or use `~/`, path that ends with or without `/`, etc.).

I think it is better to revert to using `file-in-directory-p` for now. 

(In the future, we could optimize this by caching a version of `doom-user-dir` that has been standardized using `file-truename`.)

Reply:
Yep. I have just added this.